pkg/snapshotter/snapshotter.go

```diff
@@ -73,17 +113,65 @@ func (s *snapshot) CreateSnapshot(ctx context.Context, snapshotName string, volu
 		Secrets:        snapshotterCredentials,
 	}
 
-	rsp, err := client.CreateSnapshot(ctx, &req)
-	if err != nil {
-		return "", "", time.Time{}, 0, false, err
+	var rsp *csi.CreateSnapshotResponse
+	var lastCreateError error
+	err = wait.ExponentialBackoff(*s.backoff, func() (bool, error) {
```

Comment:
This will block the whole goroutine for a long time. We did the same approach in  the CSI external provisioner and it did not work well. For example:

* When user deletes Snapshot object, the controller most probably does not need to continue in this loop.
* When the last (fifth) `CreateSnapshot` returns that the snapshot is still in progress, it won't wait until the snapshot is done (or at least I cannot check what happens, processing of `csiHandler.CreateSnapshot` return values is not included in this PR.)

Instead, we have exponential backoff in the [main controller loop itself](https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/a82cca075cb14e43242bfa0326db049deab949dd/controller/controller.go#L888) and it keeps "claims in progress" (i.e. those that were not confirmed to be finished by the driver) [separately](https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/controller/controller.go#L970), continuing to call their `Provision()` until the driver confirms it's done.